modules/images/webp-uploads/load.php

```diff
@@ -35,5 +35,186 @@ function webp_uploads_filter_image_editor_output_format( $output_format, $filena
 
 	return $output_format;
 }
-add_filter( 'image_editor_output_format', 'webp_uploads_filter_image_editor_output_format', 10, 3 );
 
+/**
+ * Hook called by `wp_generate_attachment_metadata` to create the `sources` property for every image
+ * size, the sources' property would create a new image size with all the mime types specified in
+ * `webp_uploads_valid_image_mime_types`. If the original image is one of the mimes from
+ * `webp_uploads_valid_image_mime_types` the image is just added to the `sources` property and  not
+ * created again. If the uploaded attachment is not a valid image this function does not alter the
+ * metadata of the attachment, on the other hand a `sources` property is added.
+ *
+ * @since n.e.x.t
+ *
+ * @see   wp_generate_attachment_metadata
+ * @see   webp_uploads_valid_image_mime_types
+ *
+ * @param array $metadata      An array with the metadata from this attachment.
+ * @param int   $attachment_id The ID of the attachment where the hook was dispatched.
+ *
+ * @return array An array with the updated structure for the metadata before is stored in the database.
+ */
+function webp_uploads_create_sources_property( array $metadata, $attachment_id ) {
```

Comment:
Since the `wp_generate_attachment_metadata` filter is used for any kind of attachment, we need to include a check on top of this function, to only do something if the attachment is actually an image. If not, we need to just short-circuit the callback and do nothing. It makes sense to use the same check as core does for that, see https://github.com/WordPress/wordpress-develop/blob/5.9.0/src/wp-admin/includes/image.php#L484.